sentry_streams/sentry_streams/adapters/arroyo/rust_arroyo.py

```diff
@@ -163,9 +163,23 @@ def sink(self, step: Sink, stream: Route) -> Route:
         route = RustRoute(stream.source, stream.waypoints)
 
         if isinstance(step, GCSSink):
+            if sink_config := self.steps_config.get(step.name):
+                bucket = (
+                    step.bucket if not sink_config.get("bucket") else str(sink_config.get("bucket"))
+                )
+                object_generator = (
+                    step.object_generator
+                    if not sink_config.get("object_file")
+                    else lambda: str(sink_config.get("object_file"))
+                )
```

Comment:
I think we should fail if there is a mismatch between object name generator and statically defined object name.
If the application code defines a generator, we should not allow the config to override it with a statically defined object name. That would likely break the semantics of the application.

Reply:
spunds good

Reply:
no, sorry I misunderstood that. 
IF we always provide a callable, then there should be no way to override that via config.

Reply:
Ok, I'll remove the ability to override the generators, but keep the ability to override the bucket name